pkg/api/validation/events.go

```diff
@@ -25,11 +25,19 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
This actually highlights a problem we have where references aren't normalizing kind to internal when converted from external and vice versa.  Without the version this reference appeared in, we cant distinguish between two names that mean different internal things across versions (for instance, we can't reintroduce "Minion" in v4 to mean something besides Node). For now, it's probably acceptable to make this check, but it's essentially unversioned.